### PR TITLE
remove GAed KubeletCredentialProviders （1.28）

### PR DESCRIPTION
**Pull request by pacoxu**
/kind cleanup

Fixes https://github.com/kubernetes/enhancements/issues/2133

#### Special notes for your reviewer:
- [x] https://github.com/kubernetes/test-infra/pull/29126 (wait for this to be merged.)
#### Does this PR introduce a user-facing change?

```release-note
remove GAed feature gate KubeletCredentialProviders
```

### CODE REVIEW

File: test/e2e_node/image_credential_provider.go [hunk 1]
```diff
@@ -30,7 +30,7 @@ import (
 	admissionapi "k8s.io/pod-security-admission/api"
 )
 
-var _ = SIGDescribe("ImageCredentialProvider [Feature:KubeletCredentialProviders]", func() {
```
Comment (pacoxu) [inline test/e2e_node/image_credential_provider.go:33]: The fg is GAed in v1.26. Should we remove the feature tag in e2e node test? But there are some test ci for this and we should not change it now in my mind.

- I opened  https://github.com/kubernetes/test-infra/pull/29126 to cleanup the feature gate usage only.

**Comment by pacoxu** [comment]
/retest
after https://github.com/kubernetes/test-infra/pull/29126 is merged

File: cluster/gce/gci/configure.sh [hunk 2]
```diff
@@ -717,8 +717,7 @@ function install-kube-binary-config {
   # are presented to kubelet:
   # --image-credential-provider-config=${path-to-config}
   # --image-credential-provider-bin-dir=${path-to-auth-provider-binary}
-  # Also, it is required that DisableKubeletCloudCredentialProviders and KubeletCredentialProviders
```
Comment (bart0sh) [inline cluster/gce/gci/configure.sh:720]: Why removing mention of the `DisableKubeletCloudCredentialProviders` ?

Comment (pacoxu) [reply cluster/gce/gci/configure.sh:720]: My mistake.
I added it back.

**Comment by pacoxu** [comment]
Ping @derekwaynecarr @mrunalp @dchen1107 for approval
and 
/cc @andrewsykim @adisky @ndixita
the feature owners

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm

**Review by adisky — COMMENTED** [review]
/lgtm

**Review by mrunalp — APPROVED** [review]

**Comment by dims** [comment]
/approve
/lgtm
/hold

@pacoxu let's fix the test-infra references as well ( https://cs.k8s.io/?q=KubeletCredentialProviders&i=nope&files=&excludeFiles=&repos=kubernetes/kubernetes,kubernetes/test-infra ) and i can land both PRs simultaneously

**Comment by pacoxu** [comment]
> @pacoxu let's fix the test-infra references as well ( https://cs.k8s.io/?q=KubeletCredentialProviders&i=nope&files=&excludeFiles=&repos=kubernetes/kubernetes,kubernetes/test-infra ) and i can land both PRs simultaneously

I think we can remove those test that focus on KubeletCredentialProviders as this is GAed. I removed the `[Feature:KubeletCredentialProviders]` from the test below.

https://github.com/pacoxu/kubernetes/blob/27cdddeaf16717d0115d9445128520a0db85c245/test/e2e_node/image_credential_provider.go#L33-L36

**Comment by pacoxu** [comment]
> /test pull-kubernetes-e2e-gce-kubelet-credential-provider

I removed it in https://github.com/kubernetes/test-infra/pull/29454

File: test/e2e_node/image_credential_provider.go (hunk 1 above)
Comment (BenTheElder) [reply test/e2e_node/image_credential_provider.go:33]: Is this expected to work in all clusters? Feature also implies tests that are not portable. This test is known to be problematic but I'm not sure re: portability

**Pull request merged**